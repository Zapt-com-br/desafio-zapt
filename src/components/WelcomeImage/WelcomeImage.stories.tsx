import React from 'react';

import { Story, Meta } from '@storybook/react/types-6-0';
import WelcomeImage from '.';

export default {
  title: 'Welcome/WelcomeImage',
  component: WelcomeImage,
} as Meta;

export const WelcomeImageDefault: Story = () => <WelcomeImage />;
