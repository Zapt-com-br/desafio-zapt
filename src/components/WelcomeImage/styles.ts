import styled from 'styled-components';

export const Wrapper = styled.div`
  position: relative;
  max-height: 31.3rem;
  max-width: 41.8rem;
`;

export const Image = styled.img`
  border-radius: 0 0.4rem 0.4rem 0;
  width: 100%;
  height: 100%;
`;

export const CircleImageContainer = styled.div`
  position: absolute;
  top: 1.6rem;
  left: 7.1rem;
`;

export const QuarterCircumferenceImageContainer = styled.div`
  position: absolute;
  bottom: 3.7rem;
  right: 3.8rem;
`;
