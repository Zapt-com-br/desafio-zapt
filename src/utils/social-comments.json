{
  "title": "O que est√£o falando nas redes sociais",
  "comments": [
    {
      "id": 1,
      "comment": "N√£o da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um s√≥ dia... O pessoal aqui de casa ficou viciado de um n√≠vel que terei que comprar toda semana... Assim n√£o tem academia que secura rsrsrsrs Obrigado Zapt.",
      "name": "ferri-gui",
      "image": "/images/social-comments/ferri-gui.svg"
    },
    {
      "id": 2,
      "comment": "Estava precisando achar algum lugar que vendia produtos que v√™em direto da fazenda... a Zapt salvou.. e ainda paquei um pre√ßo muito mais barato que supermecado üôå",
      "name": "paolaC",
      "image": "/images/social-comments/paola.svg"
    },
    {
      "id": 3,
      "comment": "Todos os vinhos da vin√≠cula Bueno est√£o dispon√≠veis no catalogo da Zapt a um pre√ßo incriv√©l... n√£o percam essa oportunidade...",
      "name": "Galv√£o",
      "image": "/images/social-comments/galvao.svg"
    },
    {
      "id": 4,
      "comment": "Nunca tinha achado os produtos na Lush depois que ela saiu do Brasil.. s√≥ a Zapt para conseguir importalos a pre√ßo mais barato do que vendiam aqui! Muito obrigado, recomendo sempre a todos os amigos.",
      "name": "LiciaR",
      "image": "/images/social-comments/liciar.svg"
    },
    {
      "id": 5,
      "comment": "Ontem chegou uma pizza que comprei na Zapt e me lembrou as que minha av√≥ fazia... sensacional!",
      "name": "shitz",
      "image": "/images/social-comments/shitz.svg"
    },
    {
      "id": 6,
      "comment": "Nossa cada dia tem uma oferta mais legal do que a outra.. Al√©m de produtos de qualidade o pr√©co √© imbat√≠vel",
      "name": "rafael",
      "image": "/images/social-comments/rafael.svg"
    },
    {
      "id": 7,
      "comment": "Achei uma promo√ß√£o incr√≠vel de roupinha de beb√™, foi o que salvou o enxoval do Pedro.",
      "name": "gust",
      "image": "/images/social-comments/gust.svg"
    },
    {
      "id": 8,
      "comment": "Meu dono finalmente conseguiu achar uma comida que eu gosto! Agora vou conseguir engordar e ser feliz.",
      "name": "Cookie",
      "image": "/images/social-comments/cookie.svg"
    },
    {
      "id": 9,
      "comment": "N√£o conhecia a Zapt.. um amigo meu indicou... falou para eu dar uma olhada nos pre√ßos das cervejas artesanais.. incr√≠vel.",
      "name": "duque",
      "image": "/images/social-comments/duque.svg"
    }
  ]
}
