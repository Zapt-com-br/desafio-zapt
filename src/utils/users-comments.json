{
  "title": "O que as pessoas estão achando",
  "comments": [
    {
      "id": 1,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 2,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    },
    {
      "id": 3,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 4,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    }
  ]
}
