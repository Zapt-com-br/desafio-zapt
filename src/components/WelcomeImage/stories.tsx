import { Story, Meta } from '@storybook/react/types-6-0';
import WelcomeImage from '.';

export default {
  title: 'WelcomeImage',
  component: WelcomeImage,
} as Meta;

export const Default: Story = () => <WelcomeImage />;
