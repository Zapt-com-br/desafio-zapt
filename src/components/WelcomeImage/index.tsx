import CircleImage from 'components/CircleImage';
import QuarterCircumferenceImage from 'components/QuarterCircumferenceImage';

import * as S from './styles';

const WelcomeImage = () => (
  <S.Wrapper>
    <S.Image src="/img/group.png" alt="Grupo de amigos" />

    <S.CircleImageContainer>
      <CircleImage />
    </S.CircleImageContainer>

    <S.QuarterCircumferenceImageContainer>
      <QuarterCircumferenceImage />
    </S.QuarterCircumferenceImageContainer>
  </S.Wrapper>
);

export default WelcomeImage;
