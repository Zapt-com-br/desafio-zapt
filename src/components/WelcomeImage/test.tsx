import { render, screen } from 'utils/test-utils';

import WelcomeImage from '.';

describe('<WelcomeImage />', () => {
  it('should render correctly', () => {
    const { container } = render(<WelcomeImage />);

    expect(
      screen.getByRole('img', { name: /grupo de amigos/i })
    ).toBeInTheDocument();

    expect(screen.getByRole('img', { name: /circle/i })).toBeInTheDocument();
    expect(
      screen.getByRole('img', { name: /quarter circumference/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
