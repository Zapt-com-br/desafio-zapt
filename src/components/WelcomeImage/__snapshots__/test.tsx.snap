// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeImage /> should render correctly 1`] = `
.c3 {
  color: #ffffff;
  width: 8rem;
  height: 8rem;
}

.c5 {
  color: #ffffff;
  width: 17.2rem;
  height: 10.5rem;
}

.c0 {
  position: relative;
  max-height: 31.3rem;
  max-width: 41.8rem;
}

.c1 {
  border-radius: 0 0.4rem 0.4rem 0;
  width: 100%;
  height: 100%;
}

.c2 {
  position: absolute;
  top: 1.6rem;
  left: 7.1rem;
}

.c4 {
  position: absolute;
  bottom: 3.7rem;
  right: 3.8rem;
}

<div
  class="c0"
>
  <img
    alt="Grupo de amigos"
    class="c1"
    src="/img/group.png"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <svg
        aria-label="Circle"
        fill="none"
        role="img"
        viewBox="0 0 80 80"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="40"
          cy="40"
          fill="currentColor"
          r="40"
        />
      </svg>
    </div>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <svg
        aria-label="Quarter Circumference"
        fill="none"
        role="img"
        viewBox="0 0 172 105"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M169.115 5.3L128.569.029a3.304 3.304 0 00-3.649 2.513c-1.797 7.762-8.397 29.263-28.145 42.75C77.277 58.603 49.39 61.18 13.84 52.963c-1.769-.41-3.544.658-3.979 2.418L.098 94.936a3.301 3.301 0 002.448 4.007C19.85 102.985 36.09 105 51.236 105c27.976 0 52.174-6.888 72.303-20.632 36.065-24.625 46.109-63.388 48.399-75.176.366-1.865-.936-3.645-2.823-3.891z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
